v0/destinations/singular/util.js

```diff
@@ -0,0 +1,172 @@
+ * @param {*} eventAttributes custom attributes
+ * @returns list of revenue event responses
+ */
+const generateRevenuePayloadArray = (
+  products,
+  payload,
+  Config,
+  eventAttributes
+) => {
+  const responseArray = [];
+  products.forEach(product => {
+    const productDetails = constructPayload(
+      product,
+      MAPPING_CONFIG[CONFIG_CATEGORIES.PRODUCT_PROPERTY.name]
+    );
+    let finalpayload = { ...payload, ...productDetails };
+    finalpayload = removeUndefinedAndNullValues(finalpayload);
+    const response = defaultRequestConfig();
+    response.endpoint = `${BASE_URL}/evt`;
+    response.params = { ...finalpayload, a: Config.apiKey };
+    if (eventAttributes) {
+      response.params = { ...response.params, e: eventAttributes };
+    }
+    response.method = defaultGetRequestConfig.requestMethod;
+    responseArray.push(response);
+  });
+  return responseArray;
+};
+
+const exclusionList = {
+  ANDROID_SESSION_EXCLUSION_LIST: SINGULAR_SESSION_ANDROID_EXCLUSION,
+  IOS_SESSION_EXCLUSION_LIST: SINGULAR_SESSION_IOS_EXCLUSION,
+  ANDROID_EVENT_EXCLUSION_LIST: SINGULAR_EVENT_ANDROID_EXCLUSION,
+  IOS_EVENT_EXCLUSION_LIST: SINGULAR_EVENT_IOS_EXCLUSION
+};
+
+/**
+ * Determinse if the event is a session event or not
+ * @param {*} Config
+ * @param {*} eventName
+ */
+const isSessionEvent = (Config, eventName) => {
+  const mappedSessionEvents = _.map(
+    Config.sessionEventList,
+    "sessionEventName"
+  );
+  return (
+    mappedSessionEvents.includes(eventName) ||
+    SESSIONEVENTS.includes(eventName.toLowerCase())
+  );
+};
+
+/**
+ * Based on platform of device this function generates payload for singular API
+ * @param {*} message
+ * @param {*} isSessionEvent
+ * @returns
+ */
+const platformWisePayloadGenerator = (message, isSessionEvent) => {
+  let payload;
+  let eventAttributes;
+  let platform = getValueFromMessage(message, "context.os.name");
+  const typeOfEvent = isSessionEvent ? "SESSION" : "EVENT";
+  if (!platform) {
+    throw new CustomError("[Singular] :: Platform name is missing", 400);
+  }
+  platform = platform.toLowerCase();
+  if (!SUPPORTED_PLATFORM[platform]) {
+    throw new CustomError("[Singular] :: Platform is not supported");
+  }
+
+  payload = constructPayload(
+    message,
+    MAPPING_CONFIG[
+      CONFIG_CATEGORIES[`${typeOfEvent}_${SUPPORTED_PLATFORM[platform]}`].name
+    ]
+  );
+
+  if (!payload) {
+    throw new CustomError(
+      `Failed to Create ${platform} ${typeOfEvent} Payload`,
+      400
+    );
+  }
+  if (isSessionEvent) {
+    // context.device.adTrackingEnabled = true implies Singular's do not track (dnt)
+    // to be 0 and vice-versa.
+    let adTrackingEnabled = getValueFromMessage(
+      message,
+      "context.device.adTrackingEnabled"
+    );
+    if (adTrackingEnabled === true) {
+      payload.dnt = 0;
+    } else {
+      payload.dnt = 1;
+    }
+  }
+
+  // Custom Attribues is not supported by session events
+  if (!isSessionEvent) {
```

Comment:
Can we move this as an else part of the above if

Reply:
added